fsspec/tests/abstract/__init__.py

```diff
@@ -20,55 +19,97 @@ def fs_join():
         """
         return os.path.join
 
-    @staticmethod
     @pytest.fixture
-    def fs_scenario_cp(fs, fs_join, fs_path):
+    def fs_scenario_cp(self, fs, fs_join, fs_path):
         """
         Scenario on remote filesystem that is used for many cp/get/put tests.
 
-        📁 source
-        ├── 📄 file1
-        ├── 📄 file2
-        └── 📁 subdir
-            ├── 📄 subfile1
-            ├── 📄 subfile2
-            └── 📁 nesteddir
-                └── 📄 nestedfile
+        Cleans up at the end of each test it which it is used.
         """
-        source = fs_join(fs_path, "source")
-        subdir = fs_join(source, "subdir")
-        nesteddir = fs_join(subdir, "nesteddir")
-        fs.makedirs(nesteddir)
-        fs.touch(fs_join(source, "file1"))
-        fs.touch(fs_join(source, "file2"))
-        fs.touch(fs_join(subdir, "subfile1"))
-        fs.touch(fs_join(subdir, "subfile2"))
-        fs.touch(fs_join(nesteddir, "nestedfile"))
-        return source
+        source = self._scenario_cp(fs, fs_join, fs_path)
```

Comment:
Little surprised here, because it looks like we need cp to be working correctly before we get to running the tests. Wasn't touch() more explicit?

Reply:
Looking further, I see you just moved the touch() calls. But isn't _scenario_cp misnamed, then? In general, for any of these that a downstream developer is expected to implement (if any), it would be worthwhile describing all the arguments of all the methods better. Perhaps general documentation in https://filesystem-spec.readthedocs.io/en/latest/developer.html and/or https://filesystem-spec.readthedocs.io/en/latest/copying.html would help.

Reply:
how about bulk_operations_scenario, perhaps suffixed with 0 if there might eventually be more of them?